synapseclient/__main__.py

```diff
@@ -957,13 +957,22 @@ def build_parser():
 
 def perform_main(args, syn):
     if 'func' in args:
+        if args.func != login:
+            login_with_prompt(syn, args.synapseUser, args.synapsePassword, silent=True)
         try:
             args.func(args, syn)
         except Exception as ex:
             if args.debug:
                 raise
             else:
                 sys.stderr.write(utils._synapse_error_msg(ex))
+    else:
+        # if no command provided print out help and quit
+        # if we require python 3.7 or above, we can use required argument add_subparsers
```

Comment:
I suggest tweaking this comment slightly to for clarity:

_if we require python 3.7 or above, we can use required argument tp add_subparsers instead_